### DISCUSSION TITLE
Could WinMerge show 2 identical files as different?

### DISCUSSION THREAD

**Discussion by InterWeb443** [WinMerge/winmerge#801] [Q&A]
I was unable to find anything directly related to me issue.  If I just missed the right discussion please point me to it and I can close this.

I use AOMEI Backupper to back up the Users folder (and subfolders).  Every so often I do a spot check.  The other day I decided to verify that my StudioOne folder / sub-folders / files were being backed up correctly.  Everything was identical, except for one wav file.  If you're not aware, Studio One is Digital Audio Workstation software.  This was file was one of many that I'd recorded.  There are 314 folders & 5830 files that I used WinMerge to compare.

However, the two files are identical size (62,778686 bytes).

I had started a ticket with AOMEI asking whether there were bugs in the backup software (I need to count on every file being backed up).  After some thinking, I wondered whether the error might be in WinMerge.  I can't think of any other way to verify whether these two files are different, since they are binary.

I am running WinMerge 2.16.8.0, which I just upgraded the other day.  I believe that this is the latest version.  Windows 10.0.19041 Professional.

I can upload the 2 files if you think it will help.

**Comment by sdottaka** [comment]
You can compare different files with a hex editor by selecting Compare As → Binary menu item as shown below.
You can use this to find out if the files are really different

![image](https://user-images.githubusercontent.com/98126/119205024-041f7100-bad2-11eb-80ae-214d04893356.png)
![image](https://user-images.githubusercontent.com/98126/119205394-1a79fc80-bad3-11eb-9a34-528b313fcfd2.png)

Alternatively, you can run the fc command at the command prompt to see if the files are different.

~~~
fc /b file1 file2
~~~

**Reply by InterWeb443** [reply]
Thanks.  I used the fc command and it also came up with a lot of differences.  So it appears that WinMerge is functioning correctly.

**Reply by Gitoffthelawn** [reply]
@InterWeb443 That's good to hear.  Let us know what AOMEI says.  It's concerning if a backup program does not make accurate backups!  Personally, I use backup software that also verifies the copies after they are written.

**Reply by InterWeb443** [reply]
What backup software do you use?

**Reply by Gitoffthelawn** [reply]
@InterWeb443 I use two different tools for backups:
1. Custom scripts I have written myself.  Those won't help you directly, but they are easy to write if you have coding skills.
2. The portable version of SyncBackFree from 2BrightSparks.  The portable version isn't easy to find on their site, so I'll provided this link for your convenience: https://forum.2brightsparks.com/bb/viewtopic.php?f=29&t=9641
